arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,37 @@
 #include <Smartcar.h>
 
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+const int TRIGGER_PIN = 6;
+const int ECHO_PIN = 7;
+const unsigned int MAX_DISTANCE = 300;
+const int minimumDistance = 50;
+
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+SimpleCar car(control);
+
 void setup() {
-  // put your setup code here, to run once:
+  Serial.begin(9600);
+    car.setSpeed(100);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
+  Serial.println(front.getDistance());
+  delay(10);
+
+  if (front.getDistance() <= minimumDistance && front.getDistance() > 0)
+  {
+    car.setSpeed(0);
+  }
+  else if(front.getDistance() > minimumDistance) {
```

Comment:
Every time you call `getDistance` you get a _new_ measurement. Therefore, this is potentially a source of bugs.